### PR TITLE
[CMake][Driver][SDK][Runtime] remove libunwind dependency

### PR DESCRIPTION
**Pull request by whchung**
It has been verified certain applications with complex shared library loading
sequence would have exception handlers misplaced due to libunwind. In this
commit remove libunwind dependency from HCC in different aspect:

- CMake & HCC export CMake scripts
- hcc-config
- hc_rt_debug.h
- Dockerfile

In future commits it can be considered to make this an optional feature which
is enabled at CMake config time.

### CODE REVIEW

**Comment by whchung** [comment]
@scchan as `Clang` submodule has also been changed at ( https://github.com/RadeonOpenCompute/hcc-clang-upgrade/pull/131 ), it is required to have another submodule configuration change after this PR is merged.

Also please kindly help cherry pick this commit into `roc-1.7.x` branch so it goes into the next ROCm 1.7.x release.

File: include/hc_rt_debug.h [hunk 1]
```diff
@@ -78,55 +76,7 @@ extern thread_local ShortTid hcc_tlsShortTid;
...
-      std::snprintf(loc, buffer_size, "0x%016lx:\t%s + 0x%lx\n", pc, print_func_name, offp);
-      bt += std::string(loc);
-
-      if (demangled)
-        free(demangled);
-    }
-    return bt;
-  }
-
   static void print_backtrace() {
```
Comment (scchan) [inline include/hc_rt_debug.h:79]: we should just remove this function?

**Review by scchan — CHANGES_REQUESTED** [review]
The clang PR has already been merged, could you update the link to clang with this PR?

https://github.com/RadeonOpenCompute/hcc-clang-upgrade/pull/131

File: include/hc_rt_debug.h (hunk 1 above)
Comment (whchung) [reply include/hc_rt_debug.h:79]: @scchan there are other callsites for the function so I leave it here as an empty function for now. Later on we may want to reintroduce the logic and make it an optional feature.

Comment (whchung) [reply include/hc_rt_debug.h:79]: @scchan also I added `clang` submodule into this PR

**Review by scchan — APPROVED** [review]
LGTM

**Pull request merged**